streams/src/main/java/org/apache/kafka/streams/kstream/SlidingWindowDef.java

```diff
@@ -186,7 +185,7 @@ public void flush() {
             IntegerSerializer intSerializer = new IntegerSerializer();
             ByteArraySerializer byteArraySerializer = new ByteArraySerializer();
 
-            RecordCollector collector = ((ProcessorContextImpl) context).recordCollector();
+            RecordCollector collector = ((RecordCollector.Supplier) context).recordCollector();
```

Comment:
@guozhangwang: The primary benefit is that we can use these components *in unit tests using a mock `ProcessorContext`. Without this small abstraction, all unit tests would have to use a real `ProcessorContextImpl` and everything that requires.